src/hashtable.py

```diff
@@ -74,7 +75,11 @@ def retrieve(self, key):
 
         Fill this in.
         '''
-        pass
+        index = self._hash_mod(key)
+        if pair.key in self.storage[index]:
```

Comment:
You might have a reference error here? You're thinking on the right track though..